src/algebra/quaternion.lean

```diff
@@ -131,6 +140,17 @@ by ext; simp
   (mk a₁ a₂ a₃ a₄ : ℍ[R, c₁, c₂]) - mk b₁ b₂ b₃ b₄ = mk (a₁ - b₁) (a₂ - b₂) (a₃ - b₃) (a₄ - b₄) :=
 rfl
 
+@[simp, norm_cast] lemma coe_im : (x : ℍ[R, c₁, c₂]).im = 0 := rfl
+
+@[simp] lemma re_add_im : ↑a.re + a.im = a :=
+ext _ _ (add_zero _) (zero_add _) (zero_add _) (zero_add _)
+
+@[simp] lemma sub_self_im : a - a.im = a.re :=
+ext _ _ (sub_zero _) (sub_self _) (sub_self _) (sub_self _)
+
+@[simp] lemma sub_self_re : a - a.re = a.im :=
+ext _ _ (sub_self _) (sub_zero _) (sub_zero _) (sub_zero _)
```

Comment:
```suggestion
@[simp] lemma sub_self_im : a - a.im = a.re := sub_eq_of_eq_add a.re_add_im
@[simp] lemma sub_self_re : a - a.re = a.im := sub_eq_of_eq_add' a.re_add_im
```
if you want

Reply:
Sure, why not

Reply:
Answer: because it doesn't work, the `add_group` structure isn't available yet.